cmd/acra_genauth/acra_genauth.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2018, Cossack Labs Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"flag"
+	log "github.com/sirupsen/logrus"
+	"os"
+	"github.com/cossacklabs/acra/logging"
+	"github.com/cossacklabs/acra/cmd"
+	"github.com/cossacklabs/themis/gothemis/cell"
+	"io/ioutil"
+	"strings"
+	"fmt"
+	"github.com/cossacklabs/acra/keystore"
+)
+
+type HashedPasswords map[string]string
+
+const (
+	PasswordSeparator = ":"
+	LineSeparator     = "\n"
+)
+
+func (hp HashedPasswords) Bytes() (passwordBytes []byte) {
+	passwordBytes = []byte{}
+	for name, hash := range hp {
+		passwordBytes = append(passwordBytes, []byte(name+PasswordSeparator+hash+LineSeparator)...)
+	}
+	return passwordBytes
+}
+
+func (hp HashedPasswords) WriteToFile(file string, keystore *keystore.FilesystemKeyStore) error {
+	key, err := keystore.GetAuthKey(false)
+	if err != nil {
+		return err
+	}
+	SecureCell := cell.New(key, cell.CELL_MODE_SEAL)
+	crypted, _, err := SecureCell.Protect(hp.Bytes(), nil)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(file, crypted, 0644)
+}
+
+func (hp HashedPasswords) SetPassword(name, password string) (err error) {
+	if len(password) == 0 {
+		return errors.New("passwords is empty")
+	}
+	salt := cmd.RandomStringBytes(16)
+	argon2Params := cmd.InitArgon2Params()
+	hashBytes, err := cmd.HashArgon2(password, salt, argon2Params)
+	if err != nil {
+		return err
+	}
+	if err != nil {
+		return err
+	}
+	a := cmd.UserAuth{Salt: salt, Hash: hashBytes, Argon2Params: argon2Params}
+	hp[name] = a.UserAuthString(":", ",")
```

Comment:
would be more readable if `:` and `,` will be constants like `DEFAULT_USER_PARAMS_DELIMETER` and `PARAMS_DELIMETER` because I used ctrf+f to understand how this args will be used in `UserAuthString` and for what.